### PR TITLE
Httplog

### PR DESCRIPTION
**Pull request by jdonszelmann**
Httplog is a pkg we made, and use which allows logs from the management os to be sent over http to the control server, and displayed there.

### CODE REVIEW

File: control_server/httpserver/routes.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	"github.com/google/uuid"
+
+	"baas/control_server/machines"
+	"baas/pkg/api"
+	"baas/pkg/model"
+)
+
+// Routes is a struct on which functions are defined that respond to requests
+// from the management OS. This struct holds state necessary for the request handlers.
+type Routes struct {
```
Comment (NULLx76) [inline control_server/httpserver/routes.go:25]: Maybe a constructor would be nice as that would make it easier to add more advanced things in the future

File: control_server/httpserver/routes.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	vars := mux.Vars(r)
+	id, ok := vars["uuid"]
+	if !ok || id == "" {
+		http.Error(w, "Invalid uuid", http.StatusBadRequest)
+		log.Error("Invalid uuid given")
+		return
+	}
+
+	path := fmt.Sprintf("%s/%s", routes.diskpath, id)
+	temppath := path + "." + uuid.New().String() + ".tmp"
```
Comment (NULLx76) [inline control_server/httpserver/routes.go]: use `Sprintf` for consistency? (not very important)

File: management_os/entrypoint/main.go [hunk 3]
```diff
@@ -2,24 +2,39 @@ package main
...
 
+var baseurl = fmt.Sprintf("http://control_server:%d", api.Port)
+
+func init() {
+	log.AddHook(httplog.NewLogHook(baseurl+"/log", "MMOS"))
+}
+
 func main() {
-	c := APIClient{baseURL: fmt.Sprintf("http://control_server:%d", api.Port)}
+	c := APIClient{baseURL: baseurl}
```
Comment (NULLx76) [inline management_os/entrypoint/main.go]: Again constructor, may be useful?

**Review by NULLx76 — COMMENTED** [review]
Could you also add me to the co-authors of 77be093 :)

File: control_server/httpserver/routes.go (hunk 1 above)
Comment (jdonszelmann) [reply control_server/httpserver/routes.go:25]: fixed

File: control_server/httpserver/routes.go (hunk 2 above)
Comment (jdonszelmann) [reply control_server/httpserver/routes.go]: fixed

File: management_os/entrypoint/main.go (hunk 3 above)
Comment (jdonszelmann) [reply management_os/entrypoint/main.go]: fixed

**Review by NULLx76 — APPROVED** [review]
LGTM; but depends on #20 right?

**Pull request merged**